tests/unit/niceml/utilities/test_readwritelock.py

```diff
@@ -20,79 +29,81 @@ def fs_path_config():
 
 
 def test_read_lock(fs_path_config):
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "read.lock"))
-    rw_lock = ReadLock(fs_path_config)
-    rw_lock.acquire()
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert cur_fs.exists(join_fs_path(cur_fs, root_path, "read.lock"))
+    read_lock = ReadLock(fs_path_config)
+    assert read_lock.is_acquirable()
+    read_lock.acquire()
+    assert not read_lock.is_acquirable()
     time.sleep(1)
-    rw_lock.release()
+    read_lock.release()
 
     # lock files should not exist anymore after release
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "read.lock"))
+    assert read_lock.is_acquirable()
 
 
 def test_write_lock(fs_path_config):
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
     write_lock = WriteLock(fs_path_config)
+    assert write_lock.is_acquirable()
     write_lock.acquire()
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
+    assert not write_lock.is_acquirable()
     time.sleep(1)
     write_lock.release()
 
     # lock files should not exist anymore after release
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "read.lock"))
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
+    assert write_lock.is_acquirable()
 
 
-def test_read_write_lock(fs_path_config):
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
-    write_lock = WriteLock(fs_path_config, write=True)
-    write_lock.acquire()
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
-    time.sleep(1)
-    write_lock.release()
+def test_equality(fs_path_config):
```

Comment:
Why do we need a test for checking equality?

Reply:
You are right, it is not necessary.